kore/src/Kore/Step/Simplification/Bottom.hs

```diff
@@ -11,14 +11,19 @@ module Kore.Step.Simplification.Bottom
     ( simplify
     ) where
 
-import Kore.Internal.OrPattern
-    ( OrPattern
+import Kore.Internal.Pattern
+    ( Pattern
+    )
+import qualified Kore.Internal.Pattern as Pattern
+    ( bottom
+    )
+import Kore.Internal.Variable
+    ( InternalVariable
     )
-import qualified Kore.Internal.OrPattern as OrPattern
 import Kore.Sort
 import Kore.Syntax.Bottom
 
 {-| simplifies a Bottom pattern, which means returning an always-false or.
 -}
-simplify :: Ord variable => Bottom Sort child -> OrPattern variable
-simplify Bottom {} = OrPattern.bottom
+simplify :: InternalVariable variable => Bottom Sort child -> Pattern variable
+simplify Bottom {} = Pattern.bottom
```

Comment:
We could preserve the sort here now.

Reply:
If we're going to preserve sorts, I'd like to do that in a different PR, because it's not enough to change this.

As an example, when computing `ceil(bottom)` we just remove the `ceil` and return the bottom. However, that `ceil` might have changed the sorts, so, normally, we'd need to change the predicate's sort when simplifying `ceil`s. Likely we also need to change `floor` simplification. I don't think we need to do anything about it, but there's a small chance that `Equal` simplification has the same problem.

So: not a big problem, but I'd prefer to separate these two issues (sort consistency and `or` simplification).